File: Entities/NotaCreditoDetalle.cs
namespace Entidades
{
    using System;

    public class NotaCreditoDetalle
    {
        private int _varIdNC;
        private double _varCantidad;
        private int _varIdProducto;
        private string _varDescripcion;
        private double _varPrecioUnitario;
        private double _varDescuento;
        private double _varImpuesto;
        private double _varTTdescuento;
        private double _varTTimpuesto;

        public int IdNC
        {
            get => 
                this._varIdNC;
            set => 
                this._varIdNC = value;
        }

        public double Cantidad
        {
            get => 
                this._varCantidad;
            set => 
                this._varCantidad = value;
        }

        public int Id_Producto
        {
            get => 
                this._varIdProducto;
            set => 
                this._varIdProducto = value;
        }

        public string Descripcion
        {
            get => 
                this._varDescripcion;
            set => 
                this._varDescripcion = value;
        }

        public double PrecioUnitario
        {
            get => 
                this._varPrecioUnitario;
            set => 
                this._varPrecioUnitario = value;
        }

        public double Descuento
        {
            get => 
                this._varDescuento;
            set => 
                this._varDescuento = value;
        }

        public double Impuesto
        {
            get => 
                this._varImpuesto;
            set => 
                this._varImpuesto = value;
        }

        public double Tdescuento
        {
            get => 
                this._varTTdescuento;
            set => 
                this._varTTdescuento = value;
        }

        public double Timpuesto
        {
            get => 
                this._varTTimpuesto;
            set => 
                this._varTTimpuesto = value;
        }
    }
}

File: Entidades/NotaCreditoDetalle.vb
'Continene: Id_NC,Cantidad, Id_Producto,Descripcion, PrecioUnitario, Descuento,
' Impuesto, Tdescuento, Timpuesto.
Public Class NotaCreditoDetalle

    'Variable extraída de la tabla NOTA_CREDITO y que específica a cual nc pertenece el detalle.
    Private _varIdNC As Integer
    Public Property IdNC() As Integer
        Get
            Return _varIdNC
        End Get
        Set(ByVal value As Integer)
            _varIdNC = value
        End Set
    End Property

    'Número de unidades de un producto que se le realiza lanota.
    Private _varCantidad As Double
    Public Property Cantidad() As Double
        Get
            Return _varCantidad
        End Get
        Set(ByVal value As Double)
            _varCantidad = value
        End Set
    End Property

    'Variable extraída de la tabla PRODUCTO y que específica a cual producto se le realiza la nota.
    Private _varIdProducto As Integer
    Public Property Id_Producto() As Integer
        Get
            Return _varIdProducto
        End Get
        Set(ByVal value As Integer)
            _varIdProducto = value
        End Set
    End Property

    'Descripción del producto al cual se le realiza la nota.
    Private _varDescripcion As String
    Public Property Descripcion() As String
        Get
            Return _varDescripcion
        End Get
        Set(ByVal value As String)
            _varDescripcion = value
        End Set
    End Property

    'Precio unitario del producto se le realiza la nota.
    Private _varPrecioUnitario As Double
    Public Property PrecioUnitario() As Double
        Get
            Return _varPrecioUnitario
        End Get
        Set(ByVal value As Double)
            _varPrecioUnitario = value
        End Set
    End Property

    'El descuento que se la rebará al total del producto(es un porcentaje).
    Private _varDescuento As Double
    Public Property Descuento() As Double
        Get
            Return _varDescuento
        End Get
        Set(ByVal value As Double)
            _varDescuento = value
        End Set
    End Property

    'El impuesto que se le agregará al producto(es un porcentaje).
    Private _varImpuesto As Double
    Public Property Impuesto() As Double
        Get
            Return _varImpuesto
        End Get
        Set(ByVal value As Double)
            _varImpuesto = value
        End Set
    End Property

    'El descuento total calculado por la ((Cantidad * PrecioUnitario)*Descuento%)
    Private _varTTdescuento As Double
    Public Property Tdescuento() As Double
        Get
            Return _varTTdescuento
        End Get
        Set(ByVal value As Double)
            _varTTdescuento = value
        End Set
    End Property

    'El impuesto total calculado por la ((Cantidad *  PrecioUnitario)*Impuesto%)
    Private _varTTimpuesto As Double
    Public Property Timpuesto() As Double
        Get
            Return _varTTimpuesto
        End Get
        Set(ByVal value As Double)
            _varTTimpuesto = value
        End Set
    End Property
End Class